rhine-terminal/TerminalSimple.hs

```diff
@@ -0,0 +1,100 @@
+{- | Example application for the @rhine-terminal@ library. -}
+
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+import Prelude hiding (putChar)
+
+-- rhine
+import Control.Monad.Schedule ()
+
+import Data.Text (Text)
+import qualified Data.Text as T
+import System.Terminal
+import System.Terminal.Internal
+import FRP.Rhine
+import System.IO hiding (putChar)
+
+import FRP.Rhine.Terminal (TerminalEventClock(..))
+import System.Exit (exitSuccess)
+
+-- Clocks
+
+data Input = Char Char
+           | Space | Backspace | Enter
+           | Exit
```

Comment:
```suggestion
data Input
  = Char Char
  | Space
  | Backspace
  | Enter
  | Exit
```

Reply:
There was some meaning to the groupings, but I'm not strongly attached to it.

Reply:
Ah I see. If the grouping is important, then a separate type would be in order:

```haskell
data Whitespace
  = Space
  | Backspace
  | Enter
data Input
  = Char Char
  | Whitespace Whitespace
  | Exit
```